server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -100,7 +101,8 @@
 
 public abstract class Engine implements Closeable {
 
-    public static final String SYNC_COMMIT_ID = "sync_id"; // TODO: Remove sync_id in 9.0
+    @UpdateForV9 // TODO: Remove sync_id in 9.0
```

Comment:
It would be helpful if the annotation itself allows a description string instead of having to comment it separately.

Reply:
I argued against that. Some of them could be quite long and multiline string literals are not as easy to manipulate as multiline comments. Moreover the machine is not going to do anything with the comment, it's purely for human consumption, so a comment seems more appropriate.